app/src/main/java/fr/free/nrw/commons/achievements/LevelController.java

```diff
@@ -0,0 +1,68 @@
+package fr.free.nrw.commons.achievements;
+
+import fr.free.nrw.commons.R;
+
+/**
+ * calculates the level of the user
+ */
+public class LevelController {
+
+    /**
+     * calculated the level of user and return the level information
+     * @param achievements
+     * @return
+     */
+    Level calculateLevelUp( Achievements achievements){
+        Level level  = new Level ();
+        level.setLevel(1);
+        level.setLevelStyle(R.style.LevelOne);
+        level.setMaximumUploadCount(20);
+        level.setMaximumUniqueImagesUsed(5);
+        if(achievements.getImagesUploaded() >= 100 && achievements.getUniqueUsedImages() >= 45){
+            level.setLevel(10);
+            level.setLevelStyle(R.style.LevelFive);
+        } else if (achievements.getImagesUploaded() >= 90 && achievements.getUniqueUsedImages() >= 40){
+            level.setLevel(9);
```

Comment:
Moreover, the `level` object is playing with a lot of magic numbers right now. These could very well be part of a `static` `enum` inside the `Level` object. 

```
public class Level {
// some properties
public LevelInfo levelInfo;

public enum LevelInfo {
    LEVEL_1(1, 2121, 0, 0),
    LEVEL_2(2, 2122, 20, 5),
    LEVEL_3(3, 2122, 30, 10),
    LEVEL_4(4, 2122, 40, 15);

    private int levelNumber;
    private int levelStyle;
    private int maxUniqueImages;
    private int maxUploadCount;

    LevelInfo(int levelNumber, int levelStyle, int maxUniqueImages, int maxUploadCount) {
        this.levelNumber = levelNumber;
        this.levelStyle = levelStyle;
        this.maxUniqueImages = maxUniqueImages;
        this.maxUploadCount = maxUploadCount;
    }

    public static LevelInfo from(int imagesUploaded, int uniqueImagesUsed) {
        LevelInfo level = LEVEL_1;

        for (LevelInfo levelInfo : LevelInfo.values()) {
            if (imagesUploaded > levelInfo.maxUploadCount && uniqueImagesUsed > levelInfo.maxUniqueImages) {
                level = levelInfo;
            }
        }
        return level;
    }
}


}
```

Reply:
Having an enum will eliminate magic numbers from this logic and will drastically simplify this class. Moreover, configuring the rules of a level would be much easier. 

This enum can also be used to construct the logic of calculating the current levels as shown in the example. :)

Reply:
thanks! I will start working on it 👍